packages/nx-plugin-openapi/src/executors/generate-api-lib-sources/executor.ts

```diff
@@ -19,40 +11,28 @@ export default async function runExecutor(
   const outputDir = context.workspace.projects[context.projectName].sourceRoot;
   const root = context.root;
 
-  return of(outputDir)
-    .pipe(
-      tap((outputDir) => {
-        logger.info(`Deleting outputDir ${outputDir}...`);
+  logger.info(`Deleting outputDir ${outputDir}...`);
+
+  deleteOutputDir(root, outputDir);
 
-        deleteOutputDir(root, outputDir);
+  logger.info(`Done deleting outputDir ${outputDir}.`);
 
-        logger.info(`Done deleting outputDir ${outputDir}.`);
-      }),
-      switchMap((outputDir) =>
-        from(
-          generateSources(
-            options.sourceSpecPathOrUrl,
-            options.sourceSpecUrlAuthorizationHeaders,
-            options.generator,
-            options.additionalProperties,
-            options.globalProperties,
-            options.typeMappings,
-            outputDir,
-          ),
-        ).pipe(
-          map(() => ({ success: true })),
-          catchError((error) => {
-            logger.error(error);
-            return of(error);
-          }),
-        ),
-      ),
-      catchError((error) => of(error)),
-    )
-    .toPromise<{ success: boolean }>();
+  await generateSources(
```

Comment:
Thank you 🙏 when I wrote this part I was still deep into RxJS and it made 100% sense to me, but this is better.

Did we lose some error handling though? Does it even matter in this context?

Reply:
I don't think we did lose error handling here, the rxjs seemed to catch and rethrow errors, this just throws the errors